File: 014-调整数组顺序使奇数位于偶数前面/014-reOrderArray.cpp
class Solution {
public:
    void reOrderArray(vector<int> &array) {
        for(int i=0;i<array.size();i++){
            bool tag=false;
            for(int j=0;j<array.size()-i-1;j++){
                if (array[j]%2==0 && array[j+1]%2==1){
                    swap(array[j],array[j+1]);
                    tag=true;
                }
            }
            if (!tag){
                break;
            }
        }
    }
};

File: 014-调整数组顺序使奇数位于偶数前面/014-reOrderArray.py
# -*- coding:utf-8 -*-
class Solution:
    def reOrderArray(self, array):
        # write code here
        for i in range(len(array)):
            isswap=False
            for j in range(len(array)-i-1):
                if array[j]%2==0 and array[j+1]%2==1:
                    array[j],array[j+1]=array[j+1],array[j]
                    isswap=True
            if not isswap:
                break
        return array
